tests/unit/test_jobs.py

```diff
@@ -3121,3 +3129,106 @@ def actor(self):
+                "src": [".packit.yaml"],
+            },
+        ],
+        paths=["."],
+        dist_git_base_url="https://src.fedoraproject.org/",
+        upstream_package_name="teamcity-messages",
+        archive_root_dir_template="{upstream_pkg_name}-{version}",
+        upstream_project_url="https://github.com/majamassarini/teamcity-messages",
+        config_file_path=".packit.yaml",
+        patch_generation_patch_id_digits=4,
+        dist_git_namespace="rpms",
+        downstream_package_name="python-teamcity-messages",
+        upstream_tag_template="v{version}",
+    )
+    python_teamcity_messages_double = copy.deepcopy(python_teamcity_messages)
+    python_teamcity_messages_double.downstream_package_name = "a double"
+
+    packages = {
+        "python-teamcity-messages": copy.deepcopy(python_teamcity_messages),
+        "python-teamcity-messages-double": python_teamcity_messages_double,
+    }
+    jobs = [
+        JobConfig(
+            type=JobType.propose_downstream,
+            trigger=JobConfigTriggerType.release,
+            skip_build=False,
+            packages={
+                "python-teamcity-messages": python_teamcity_messages,
+                "python-teamcity-messages-double": python_teamcity_messages_double,
+            },
+        ),
+        JobConfig(
+            type=JobType.propose_downstream,
+            trigger=JobConfigTriggerType.release,
+            skip_build=False,
+            packages={
+                "python-teamcity-messages": python_teamcity_messages,
+            },
+        ),
+        JobConfig(
+            type=JobType.propose_downstream,
+            trigger=JobConfigTriggerType.release,
+            skip_build=False,
+            packages={
+                "python-teamcity-messages-double": python_teamcity_messages_double
+            },
+        ),
+    ]
+    packages_config = PackageConfig(packages=packages, jobs=jobs)
+
+    class Event(ReleaseEvent):
+        def __init__(self):
+            self._package_config_searched = None
+            self._package_config = packages_config
+            self._project = flexmock()
+            self._base_project = flexmock()
+            self._pr_id = flexmock()
+            self._commit_sha = flexmock()
+            self.fail_when_config_file_missing = True
+
+        @property
+        def job_config_trigger_type(self):
+            return JobConfigTriggerType.release
+
+    event = Event()
+    steve = SteveJobs(event)
+    handlers = steve.get_handlers_for_event()
+    assert handlers
+
+    for handler in handlers:
+        job_configs = steve.get_config_for_handler_kls(handler)
+        assert job_configs
+
+        original_ref = 0
+        double_ref = 0
+        for job_config in job_configs:
+            if job_config.downstream_package_name == "a double":
+                double_ref += 1
+            else:
+                original_ref += 1
+
+        assert original_ref == 2
+        assert double_ref == 2
```

Comment:
I'd say we should also follow here and check if all jobs for all packages were processed correctly, something like:
```python
for package in packages_config.packages:
    for job in package.jobs:
        assert something(handlers) == other_thing(job)
```